app/pulp/app/models/task.py

```diff
@@ -205,6 +206,22 @@ def set_failed(self, exc, einfo):
         self.result = exception_to_dict(exc, einfo.traceback)
         self.save()
 
+    def cancel(self):
+        """
+        Set the task to a canceled state if it is in an incomplete state.
+
+        This updates the :attr: `finished_at` attribute, sets the :attr: `state` to
+        :attr: `CANCELED`
+
```

Comment:
Documenting the exception is a good idea. Under old (but still current AFAIK) way, that looks like

```
:raises ValidationError: this is why
```